packages/itwin/tree-widget/src/components/trees/common/ReportingTreeEventHandler.ts

```diff
@@ -0,0 +1,89 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import { concatMap, Observable as RxjsObservable, tap } from "rxjs";
+import { TreeEventHandler } from "@itwin/components-react";
+
+import type {
+  AbstractTreeNodeLoaderWithProvider,
+  TreeCheckboxStateChangeEventArgs,
+  TreeNodeEventArgs,
+  TreeSelectionChange,
+  TreeSelectionModificationEventArgs,
+  TreeSelectionReplacementEventArgs,
+} from "@itwin/components-react";
+import type { IPresentationTreeDataProvider } from "@itwin/presentation-components";
+export interface ReportingTreeEventHandlerProps {
+  nodeLoader: AbstractTreeNodeLoaderWithProvider<IPresentationTreeDataProvider>;
+  eventHandler: TreeEventHandler;
+  reportUsage?: (props: { reportInteraction: boolean }) => void;
+}
+
+export class ReportingTreeEventHandler extends TreeEventHandler {
+  protected _eventHandler: TreeEventHandler;
+  protected _reportUsage?: (props: { reportInteraction: boolean }) => void;
```

Comment:
I this `_reportUsage` should not be optional. It is always provided and I think it does not make sense using `ReportingTreeEventHandler` without `_reportUsage`  callback.